### PR TITLE
Refacter lookup plugins

### PR DESCRIPTION
**Pull request by tremble**
Depends-On: https://github.com/ansible-collections/amazon.aws/pull/1248

##### SUMMARY

- Refacters the lookup plugins to use common code for common boto3/botocore operations

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/lookup/aws_account_attribute.py
plugins/lookup/aws_secret.py
plugins/lookup/aws_ssm.py
plugins/module_utils/botocore.py
plugins/module_utils/core.py
plugins/module_utils/exceptions.py
plugins/module_utils/modules.py

### CODE REVIEW

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
recheck

File: meta/runtime.yml [hunk 1]
```diff
@@ -126,3 +126,9 @@ plugin_routing:
     execute_lambda:
       # Deprecation for this alias should not *start* prior to 2024-09-01
       redirect: amazon.aws.lambda_execute
+  lookup:
+    aws_ssm:
```
[the remark this answers is not in the document]
Comment (tremble) [reply meta/runtime.yml:130]: Nope, I missed it :)

[the remark this answers is not in the document]
Comment (tremble) [reply meta/runtime.yml:130]: Added a changelog entry - note deprecation cycle has not started yet, same as with the other mass-renames.

File: changelogs/fragments/1225-refacter-lookup.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+minor_changes:
+- aws_secret - the ``aws_secret`` lookup plugin has been renamed ``secretsmanager_secret``, ``aws_secret`` remains as an alias
+  (https://github.com/ansible-collections/amazon.aws/pull/1225).
+- aws_ssm - the ``aws_ssm`` lookup plugin has been renamed ``ssm_parameter``, ``aws_ssm`` remains as an alias
+  (https://github.com/ansible-collections/amazon.aws/pull/1225).
+
+- aws_account_attribute - the ``aws_account_attribute`` lookup plugin has been refactered to use
```
Comment (alinabuzachis) [inline changelogs/fragments/1225-refacter-lookup.yml:7]: Could you please double-check for this typo?

```suggestion
- aws_account_attribute - the ``aws_account_attribute`` lookup plugin has been refactored to use
```

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**